### DISCUSSION TITLE
Navigation API

### DISCUSSION THREAD

**Discussion by zkulbeda** [Devographics/surveys#175] [State of HTML Features]
Initiate, intercept, and manage navigation actions, examine history entries. Successor to the older `history` and `location` APIs.

### Code sample

```js
navigation.navigate(url, {history: "push"});
navigation.addEventListener("navigate", interceptNavigation);
```

```js
navigation.addEventListener("navigate", e => {
  if (!e.canIntercept || e.hashChange || e.downloadRequest !== null) {
    return;
  }

  if (routesTable.has(e.destination.url)) {
    const routeHandler = routesTable.get(e.destination.url);
    e.intercept({ handler: routeHandler });
  }
});
```

### MDN URL:

https://developer.mozilla.org/en-US/docs/Web/API/Navigation_API

### Specification URL:

https://wicg.github.io/navigation-api/

### Any other links?

Explainer: https://github.com/WICG/navigation-api

### How stable is this feature?

Cutting edge (Shipped, but not in all browsers)

### Type of feature

JS API

### What focus area does this feature fall under?

Interactivity, Bridging the gap with native

### Details

The API was merged into html spec recently, but google was shipped this a long time ago. 
Questions like: Did you use it? Do you waiting for shipping in all major browsers? Do you need a group- or stack-based navigation as same as native apps have?

**Comment by atopal** [comment]
Agreed, this would be a great one to include. We know that frameworks in particular care about this one.

**Comment by atopal** [comment]
@LeaVerou what do you think? Can we include this?
